pkg/webhook/shoot/mutator.go

```diff
@@ -41,6 +74,15 @@ func (m *mutator) Mutate(ctx context.Context, new, _ client.Object) error {
 			extensionswebhook.LogMutation(logger, x.Kind, x.Namespace, x.Name)
 			return m.mutateVPNShootDeployment(ctx, x)
 		}
+	case *corev1.Secret:
+		// TODO: remove this once gardener-node-agent is in use
+		// the purpose of this hack is to enable the cloud-config-downloader to pull the hyperkube image from
+		// a registry mirror in case this shoot cluster is configured with networkaccesstype restricted/forbidden
+		err = m.mutateCloudConfigDownloaderHyperkubeImage(ctx, x)
+		if err != nil {
+			// FIXME is this correct
+			m.logger.Error(err, "mutation did not work")
```

Comment:
Why not return the error if the mutation does not work? The cluster worker won't come up anyway?